server/client.go

```diff
@@ -709,6 +725,7 @@ func (c *client) deliverMsg(sub *subscription, mh, msg []byte) {
 
 	deadlineSet := false
 	if client.bw.Available() < (len(mh) + len(msg) + len(CR_LF)) {
+		atomic.AddInt64(&c.cache.wfc, 1)
```

Comment:
And if so, then why use atomic since in `deliverMsg`, client lock is held, then in readLoop, cp (which is a client) lock is held. And why is `wfc` in the cache then? Again, it looks like you originally meant to notify the client in the readLoop that a client that we are sending to need to have its buffer checked.